sdk/storage/Azure.Storage.Files.DataLake/src/PathClient.cs

```diff
@@ -0,0 +1,1462 @@
+        /// <param name="destConditions">
+        /// Optional <see cref="DataLakeRequestConditions"/> to add
+        /// conditions on the creation of this file or directory.
+        /// </param>
+        /// <param name="sourceConditions">
+        /// Optional <see cref="DataLakeRequestConditions"/> to add
+        /// conditions on the source on the creation of this file or directory.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response{PathInfo}"/> describing the
+        /// newly created page blob.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        [ForwardsClientCalls]
+        public virtual async Task<Response<PathClient>> RenameAsync(
+            string destinationPath,
+            DataLakeRequestConditions destConditions = default,
+            DataLakeRequestConditions sourceConditions = default,
+            CancellationToken cancellationToken = default) =>
+            await RenameInternal(
+                destinationPath,
+                destConditions,
+                sourceConditions,
+                true, // async
+                cancellationToken)
+                .ConfigureAwait(false);
+
+        /// <summary>
+        /// The <see cref="RenameInternal"/> operation renames a file or directory.
+        ///
+        /// For more information, see https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/create.
+        /// </summary>
+        /// <param name="destinationPath">
+        /// The destination path to rename the path to.
+        /// </param>
+        /// <param name="destConditions">
+        /// Optional <see cref="DataLakeRequestConditions"/> to add
+        /// conditions on the creation of this file or directory.
+        /// </param>
+        /// <param name="sourceConditions">
+        /// Optional <see cref="DataLakeRequestConditions"/> to add
+        /// conditions on the source on the creation of this file or directory.
+        /// </param>
+        /// <param name="async">
+        /// Whether to invoke the operation asynchronously.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response{PathInfo}"/> describing the
+        /// newly created page blob.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        private async Task<Response<PathClient>> RenameInternal(
```

Comment:
This is a new pattern we haven't really seen before and should be discussed in the API review

Reply:
How else can we do this in a way that is convenient for the customer?  Unless something better is proposed, I'm going to argue strongly for this pattern.

Reply:
Looks like this pattern was approved.